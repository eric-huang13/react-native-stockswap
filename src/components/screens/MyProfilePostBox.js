import React, {Component} from 'react';
import {
  StyleSheet,
  Image,
  TouchableOpacity,
  SafeAreaView,
  Dimensions,
} from 'react-native';
import {connect} from 'react-redux';

class MyProfilePosts extends Component {
  constructor(props) {
    super(props);

    this.state = {
      filteredComments:[]
    };
  }
   
  componentDidMount() {
    //fetch data()
    const {comments, item} = this.props;
    const filteredComments = comments.filter(
      (comment) => comment.postId === item.id,
    );
    {
        this.setState({
          filteredComments: filteredComments,
        });
    }
  }

  componentDidUpdate(prevProps) {
    const {comments, item} = this.props;
    if (this.props.comments !== prevProps.comments) {
      const filteredComments = comments.filter(
        (comment) => comment.postId === item.id,
      );
      this.setState({
        filteredComments: filteredComments,
      });
    }
  }
  render() {
    const {reply, userAccount, item} = this.props;

    return (
      <SafeAreaView style={style.container}>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate({
              name: 'PostScreen',
              params: {post: item, filteredComments: this.state.filteredComments, reply, userAccount},
            })
          }>
          <Image style={style.image} source={{uri: item.img}} />
        </TouchableOpacity>
      </SafeAreaView>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    post: state.posts.posts,
    comments: state.posts.comments,
    reply: state.posts.reply,
    userAccount: state.user.userFakeData,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MyProfilePosts);

const style = StyleSheet.create({
  container: {
    width: Dimensions.get('window').width / 3 - 2,
    margin: 1,
  },
  image: {
    height: Dimensions.get('window').width / 3 - 2,
    width: '100%',
  },
});
