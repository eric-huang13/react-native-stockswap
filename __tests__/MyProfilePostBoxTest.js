import React from 'react';
import MyProfilePostBox from '../src/components/MyProfileTabComponents/MyProfilePostBox';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<MyProfilePostBox item={{img:'test' }}/>);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});