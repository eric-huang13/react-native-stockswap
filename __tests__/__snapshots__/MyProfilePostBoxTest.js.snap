// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "margin": 1,
      "width": 248,
    }
  }
>
  <ForwardRef
    onPress={[Function]}
  >
    <Image
      source={
        Object {
          "uri": "test",
        }
      }
      style={
        Object {
          "height": 248,
          "width": "100%",
        }
      }
    />
  </ForwardRef>
</ForwardRef(SafeAreaView)>
`;
